x-pack/platform/plugins/shared/streams_app/public/components/stream_detail_overview/components/stream_stats_panel.tsx

```diff
@@ -0,0 +1,209 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import {
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiIconTip,
+  EuiPanel,
+  EuiText,
+  useEuiTheme,
+} from '@elastic/eui';
+import { css } from '@emotion/css';
+import { i18n } from '@kbn/i18n';
+import React, { ReactNode } from 'react';
+import { IngestStreamGetResponse, IngestStreamLifecycleILM } from '@kbn/streams-schema';
+import { IlmLocatorParams } from '@kbn/index-lifecycle-management-common-shared';
+
+import { LocatorPublic } from '@kbn/share-plugin/public';
+import type { StreamDetailsResponse } from '@kbn/streams-plugin/server/routes/internal/streams/crud/route';
+import { IlmLink } from '../../data_management/stream_detail_lifecycle/ilm_link';
+import {
+  formatBytes,
+  formatIngestionRate,
+} from '../../data_management/stream_detail_lifecycle/helpers/format_bytes';
+import { DataStreamStats } from '../../data_management/stream_detail_lifecycle/hooks/use_data_stream_stats';
+
+interface StreamStatsPanelProps {
+  definition?: IngestStreamGetResponse;
+  dataStreamStats?: DataStreamStats;
+  docCount?: StreamDetailsResponse;
+  ilmLocator?: LocatorPublic<IlmLocatorParams>;
+}
+
+const RetentionDisplay = ({
+  definition,
+  ilmLocator,
+}: {
+  definition?: IngestStreamGetResponse;
+  ilmLocator?: LocatorPublic<IlmLocatorParams>;
+}) => {
+  if (!definition) return <>-</>;
+
+  if ('dsl' in definition.effective_lifecycle) {
+    return (
+      <>
+        {definition?.effective_lifecycle.dsl.data_retention ||
+          i18n.translate('xpack.streams.entityDetailOverview.unlimited', {
+            defaultMessage: 'Keep indefinitely',
+          })}
+      </>
+    );
+  }
+
+  return (
+    <IlmLink
+      lifecycle={definition.effective_lifecycle as IngestStreamLifecycleILM}
+      ilmLocator={ilmLocator}
+    />
+  );
+};
+
+interface StatItemProps {
+  label: ReactNode;
+  value: ReactNode;
+  withBorder?: boolean;
+}
+
+const StatItem = ({ label, value, withBorder = false }: StatItemProps) => {
+  const { euiTheme } = useEuiTheme();
+
+  const borderStyle = withBorder
+    ? css`
+        border-left: 1px solid ${euiTheme.colors.borderBaseSubdued};
+        padding-left: ${euiTheme.size.s};
+      `
+    : '';
+
+  return (
+    <EuiFlexItem grow className={borderStyle}>
+      <EuiFlexGroup direction="column" gutterSize="xs">
+        <EuiText size="xs" color="subdued">
+          {label}
+        </EuiText>
+        <EuiText
+          size="m"
+          className={css`
+            font-weight: bold;
+          `}
+        >
+          {value}
+        </EuiText>
+      </EuiFlexGroup>
+    </EuiFlexItem>
+  );
+};
+
+function formatNumber(val: number) {
```

Comment:
I think you forgot to remove this one, since you have now `x-pack/platform/plugins/shared/streams_app/public/components/stream_detail_overview/utils/format_helpers.ts`